File: park-admin-webapp/src/app/model/estacionamiento.ts
import {Nivel} from './nivel';

export class Estacionamiento {
  nombre: string;
  id: string;
  niveles: Nivel[];

  constructor(nombre: string, id: string, niveles: Nivel[]) {
    this.nombre = nombre;
    this.id = id;
    this.niveles = niveles;
  }
}

File: park-admin-app/park-admin-server/src/main/java/com/parkadmin/parkadminserver/model/Estacionamiento.java
package com.parkadmin.parkadminserver.model;

import lombok.Data;

import javax.persistence.*;
import java.util.List;

@Data
@Entity
public class Estacionamiento {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private String nombre;
    @OneToMany(mappedBy = "nivel", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Nivel> niveles;
}
